### ISSUE TITLE
bug in pci_find_cap_nr

### ISSUE THREAD

**Issue by joevt** [pciutils/pciutils#19]
setpci is unable to find a capability unless you include the index of the capability in the @ parameter or the capability is the first capability. index is incorrectly counting all capabilities instead of the one we want. The loop should look like this:

```
  for (c=d->first_cap; c; c=c->next)
    {
      if (c->type == type && c->id == id)
	{
	  if (target == index)
	    found = c;
          index++;
	}
    }
```

**Comment by JohnAZoidberg** [comment]
Yes, that's correct. There's already a pull-request that solves this: #18
